modules/modes/roamer_reset.py

```diff
@@ -182,6 +199,8 @@ def run_emerald(self, has_good_ability: bool):
             else:
                 yield from navigate_to(MapRSE.LITTLEROOT_TOWN_BRENDANS_HOUSE_1F, (8, 8))
 
+            yield from walk_one_tile("Down")
+
```

Comment:
Ah yes, I think I had added that because I didn't realize that the new navigate_to() function operated the way it did and automatically went down an additional tile for a map transition. I checked the code of the original function and forgot to delete that line from run_emerald() (I had also added it to run_rs() at the time) after I understood how it worked.

Also, I forgot that I had written two conditions for the mode to be selected. I hadn't tested them until now, but I can actually reduce it to one, since LATIOS_OR_LATIAS_ROAMING is dependent upon SYS_TV_LATI being set. Given that that flag is set after beating the Elite Four, do I need to add any other conditions to its activation? Or will SYS_TV_LATI be sufficient on its own (with the error text that the TV should be flashing after beating the Elite Four)?

Reply:
Also, please note that I added an additional line in the determination for having a good ability to lead the party with. Arena Trap only works in Emerald in this generation.

Source: https://bulbapedia.bulbagarden.net/wiki/Arena_Trap_(Ability)#Outside_of_battle